extensions/parquet/compression/src/main/java/io/deephaven/parquet/compress/LZ4WithLZ4RawBackupCompressorAdapter.java

```diff
@@ -12,12 +12,22 @@
 import java.io.InputStream;
 
 /**
- * This is the default adapter for LZ4 data. It attempts to decompress with LZ4, and falls back to LZ4_RAW on failure.
- * The fallback mechanism is particularly useful for decompressing parquet files that are compressed with LZ4_RAW but
- * tagged as LZ4 in the metadata.
+ * This is the default adapter for LZ4 files. It attempts to decompress with LZ4 and falls back to LZ4_RAW on failure.
+ * After that, it always uses LZ4_RAW for decompression. This fallback mechanism is particularly useful for
+ * decompressing parquet files that are compressed with LZ4_RAW but tagged as LZ4 in the metadata. This adapter is
+ * internally stateful in some cases and therefore a single instance should not be re-used across files.
```

Comment:
Good comment, but I would like a note on the code that builds one of these, too.